Use DeepAR for second-level data

When using `DeepAREstimator` for second-level data, both `get_lags_for_frequency` and `time_features_from_frequency_str` do not support `freq = 'S'`. It is possible to support second-level data in these two functions? Or is it possible to disable `lags_seq` and `time_features` in `DeepAREstimator`?

Hi @lostella, 
`gluonts==0.5.0`
I followed the previous post by setting `lag_seq=[0]` in the training and the model is trained successfully while the in the evaluation using `backtest.backtest_metrics`, the following error shows up:
```
.local/lib/python3.6/site-packages/gluonts/model/deepar/_network.py in get_lagged_subsequences(F, sequence, sequence_length, indices, subsequences_length)
    150             f"{sequence_length}"
    151         )
--> 152         assert all(lag_index >= 0 for lag_index in indices)
    153 
    154         lagged_values = []

AssertionError: 
```
Is there a plan to enable the 0 lag training?

--- a/src/gluonts/time_feature/__init__.py
+++ b/src/gluonts/time_feature/__init__.py
@@ -25,11 +25,13 @@ from ._base import (
     MinuteOfHourIndex,
     MonthOfYear,
     MonthOfYearIndex,
-    norm_freq_str,
-    time_features_from_frequency_str,
+    SecondOfMinute,
+    SecondOfMinuteIndex,
     TimeFeature,
     WeekOfYear,
     WeekOfYearIndex,
+    norm_freq_str,
+    time_features_from_frequency_str,
 )
 from .holiday import SPECIAL_DATE_FEATURES, SpecialDateFeatureSet
 from .lag import get_lags_for_frequency
@@ -51,6 +53,8 @@ __all__ = [
     "MinuteOfHourIndex",
     "MonthOfYear",
     "MonthOfYearIndex",
+    "SecondOfMinute",
+    "SecondOfMinuteIndex",
     "norm_freq_str",
     "SPECIAL_DATE_FEATURES",
     "SpecialDateFeatureSet",

--- a/src/gluonts/time_feature/_base.py
+++ b/src/gluonts/time_feature/_base.py
@@ -37,6 +37,24 @@ class TimeFeature:
         return self.__class__.__name__ + "()"
 
 
+class SecondOfMinute(TimeFeature):
+    """
+    Second of minute encoded as value between [-0.5, 0.5]
+    """
+
+    def __call__(self, index: pd.PeriodIndex) -> np.ndarray:
+        return index.second.values / 59.0 - 0.5
+
+
+class SecondOfMinuteIndex(TimeFeature):
+    """
+    Second of minute encoded as zero-based index, between 0 and 59.
+    """
+
+    def __call__(self, index: pd.PeriodIndex) -> np.ndarray:
+        return index.second.astype(float).values
+
+
 class MinuteOfHour(TimeFeature):
     """
     Minute of hour encoded as value between [-0.5, 0.5]
@@ -228,6 +246,14 @@ def time_features_from_frequency_str(freq_str: str) -> List[TimeFeature]:
             DayOfMonth,
             DayOfYear,
         ],
+        offsets.Second: [
+            SecondOfMinute,
+            MinuteOfHour,
+            HourOfDay,
+            DayOfWeek,
+            DayOfMonth,
+            DayOfYear,
+        ],
     }
 
     offset = to_offset(freq_str)
@@ -251,5 +277,6 @@ def time_features_from_frequency_str(freq_str: str) -> List[TimeFeature]:
         H   - hourly
         T   - minutely
             alias: min
+        S   - secondly
     """
     raise RuntimeError(supported_freq_msg)

--- a/src/gluonts/time_feature/lag.py
+++ b/src/gluonts/time_feature/lag.py
@@ -53,6 +53,12 @@ def get_lags_for_frequency(
 
     # Lags are target values at the same `season` (+/- delta) but in the
     # previous cycle.
+    def _make_lags_for_second(multiple, num_cycles=3):
+        # We use previous ``num_cycles`` hours to generate lags
+        return [
+            _make_lags(k * 60 // multiple, 2) for k in range(1, num_cycles + 1)
+        ]
+
     def _make_lags_for_minute(multiple, num_cycles=3):
         # We use previous ``num_cycles`` hours to generate lags
         return [
@@ -121,6 +127,15 @@ def get_lags_for_frequency(
             + _make_lags_for_hour(offset.n / 60.0)
             + _make_lags_for_day(offset.n / (60.0 * 24))
             + _make_lags_for_week(offset.n / (60.0 * 24 * 7))
+        )
+    # second
+    elif offset_name == "S":
+        lags = (
+            _make_lags_for_second(offset.n)
+            + _make_lags_for_minute(offset.n / 60.0)
+            + _make_lags_for_hour(offset.n / (60.0 * 60.0))
+            + _make_lags_for_day(offset.n / (60.0 * 60.0 * 24))
+            + _make_lags_for_week(offset.n / (60.0 * 60.0 * 24 * 7))
         )
     else:
         raise Exception("invalid frequency")

--- a/src/gluonts/time_feature/lag.py
+++ b/src/gluonts/time_feature/lag.py
@@ -134,7 +134,6 @@ def get_lags_for_frequency(
             _make_lags_for_second(offset.n)
             + _make_lags_for_minute(offset.n / 60)
             + _make_lags_for_hour(offset.n / (60 * 60))
-            + _make_lags_for_day(offset.n / (60 * 60 * 24))
         )
     else:
         raise Exception("invalid frequency")

--- a/src/gluonts/time_feature/lag.py
+++ b/src/gluonts/time_feature/lag.py
@@ -117,25 +117,24 @@ def get_lags_for_frequency(
     elif offset_name == "H":
         lags = (
             _make_lags_for_hour(offset.n)
-            + _make_lags_for_day(offset.n / 24.0)
-            + _make_lags_for_week(offset.n / (24.0 * 7))
+            + _make_lags_for_day(offset.n / 24)
+            + _make_lags_for_week(offset.n / (24 * 7))
         )
     # minutes
     elif offset_name == "T":
         lags = (
             _make_lags_for_minute(offset.n)
-            + _make_lags_for_hour(offset.n / 60.0)
-            + _make_lags_for_day(offset.n / (60.0 * 24))
-            + _make_lags_for_week(offset.n / (60.0 * 24 * 7))
+            + _make_lags_for_hour(offset.n / 60)
+            + _make_lags_for_day(offset.n / (60 * 24))
+            + _make_lags_for_week(offset.n / (60 * 24 * 7))
         )
     # second
     elif offset_name == "S":
         lags = (
             _make_lags_for_second(offset.n)
-            + _make_lags_for_minute(offset.n / 60.0)
-            + _make_lags_for_hour(offset.n / (60.0 * 60.0))
-            + _make_lags_for_day(offset.n / (60.0 * 60.0 * 24))
-            + _make_lags_for_week(offset.n / (60.0 * 60.0 * 24 * 7))
+            + _make_lags_for_minute(offset.n / 60)
+            + _make_lags_for_hour(offset.n / (60 * 60))
+            + _make_lags_for_day(offset.n / (60 * 60 * 24))
         )
     else:
         raise Exception("invalid frequency")

--- a/src/gluonts/time_feature/seasonality.py
+++ b/src/gluonts/time_feature/seasonality.py
@@ -21,11 +21,11 @@ logger = logging.getLogger(__name__)
 
 
 DEFAULT_SEASONALITIES = {
-    "S": 86400,
-    "T": 1440,
-    "H": 24,
-    "D": 1,
-    "W": 1,
+    "S": 3600,  # 1 hour
+    "T": 1440,  # 1 day
+    "H": 24,  # 1 day
+    "D": 1,  # 1 day
+    "W": 1,  # 1 week
     "M": 12,
     "B": 5,
     "Q": 4,

--- a/src/gluonts/time_feature/seasonality.py
+++ b/src/gluonts/time_feature/seasonality.py
@@ -14,12 +14,14 @@
 import logging
 
 import pandas as pd
+
 from gluonts.time_feature import norm_freq_str
 
 logger = logging.getLogger(__name__)
 
 
 DEFAULT_SEASONALITIES = {
+    "S": 86400,
     "T": 1440,
     "H": 24,
     "D": 1,

--- a/test/time_feature/test_agg_lags.py
+++ b/test/time_feature/test_agg_lags.py
@@ -74,7 +74,7 @@ valid_lags_rolling = {
 def test_agg_lags(pred_length):
     # create dummy dataset
     target = np.array([1, 1, 1, 2, 2, 3, 3, 4, 5, 6])
-    start = pd.Timestamp("01-01-2019 01:00:00", freq="1H")
+    start = pd.Period("01-01-2019 01:00:00", freq="1H")
     freq = "1H"
     ds = ListDataset(
         [{FieldName.TARGET: target, FieldName.START: start}], freq=freq

--- a/test/time_feature/test_features.py
+++ b/test/time_feature/test_features.py
@@ -16,22 +16,24 @@ import pandas as pd
 import pytest
 
 from gluonts.time_feature import (
-    TimeFeature,
-    MinuteOfHour,
-    HourOfDay,
-    DayOfWeek,
-    DayOfMonth,
-    DayOfYear,
-    WeekOfYear,
-    MonthOfYear,
     Constant,
-    MinuteOfHourIndex,
-    HourOfDayIndex,
-    DayOfWeekIndex,
+    DayOfMonth,
     DayOfMonthIndex,
+    DayOfWeek,
+    DayOfWeekIndex,
+    DayOfYear,
     DayOfYearIndex,
-    WeekOfYearIndex,
+    HourOfDay,
+    HourOfDayIndex,
+    MinuteOfHour,
+    MinuteOfHourIndex,
+    MonthOfYear,
     MonthOfYearIndex,
+    SecondOfMinute,
+    SecondOfMinuteIndex,
+    TimeFeature,
+    WeekOfYear,
+    WeekOfYearIndex,
 )
 
 
@@ -39,21 +41,42 @@ from gluonts.time_feature import (
     "feature, index",
     [
         (
+            SecondOfMinute(),
+            pd.period_range(
+                "01-01-2015 00:00:00", periods=60 * 2 * 24, freq="5S"
+            ),
+        ),
+        (
             MinuteOfHour(),
-            pd.date_range(
+            pd.period_range(
                 "01-01-2015 00:00:00", periods=60 * 2 * 24, freq="1min"
             ),
         ),
         (
             HourOfDay(),
-            pd.date_range("01-01-2015 00:00:00", periods=14 * 24, freq="1h"),
+            pd.period_range("01-01-2015 00:00:00", periods=14 * 24, freq="1h"),
         ),
-        (DayOfWeek(), pd.date_range("01-01-2015", periods=365 * 5, freq="D")),
-        (DayOfMonth(), pd.date_range("01-01-2015", periods=365 * 5, freq="D")),
-        (DayOfYear(), pd.date_range("01-01-2015", periods=365 * 5, freq="D")),
-        (WeekOfYear(), pd.date_range("01-01-2015", periods=53 * 5, freq="W")),
-        (MonthOfYear(), pd.date_range("01-01-2015", periods=12 * 5, freq="M")),
-        (Constant(), pd.date_range("01-01-2015", periods=5, freq="A")),
+        (
+            DayOfWeek(),
+            pd.period_range("01-01-2015", periods=365 * 5, freq="D"),
+        ),
+        (
+            DayOfMonth(),
+            pd.period_range("01-01-2015", periods=365 * 5, freq="D"),
+        ),
+        (
+            DayOfYear(),
+            pd.period_range("01-01-2015", periods=365 * 5, freq="D"),
+        ),
+        (
+            WeekOfYear(),
+            pd.period_range("01-01-2015", periods=53 * 5, freq="W"),
+        ),
+        (
+            MonthOfYear(),
+            pd.period_range("01-01-2015", periods=12 * 5, freq="M"),
+        ),
+        (Constant(), pd.period_range("01-01-2015", periods=5, freq="A")),
     ],
 )
 def test_feature_normalized_bounds(
@@ -69,40 +92,47 @@ def test_feature_normalized_bounds(
     "feature, index, cardinality",
     [
         (
+            SecondOfMinuteIndex(),
+            pd.period_range(
+                "01-01-2015 00:00:00", periods=60 * 2 * 24, freq="1S"
+            ),
+            60,
+        ),
+        (
             MinuteOfHourIndex(),
-            pd.date_range(
+            pd.period_range(
                 "01-01-2015 00:00:00", periods=60 * 2 * 24, freq="1min"
             ),
             60,
         ),
         (
             HourOfDayIndex(),
-            pd.date_range("01-01-2015 00:00:00", periods=14 * 24, freq="1h"),
+            pd.period_range("01-01-2015 00:00:00", periods=14 * 24, freq="1h"),
             24,
         ),
         (
             DayOfWeekIndex(),
-            pd.date_range("01-01-2015", periods=365 * 5, freq="D"),
+            pd.period_range("01-01-2015", periods=365 * 5, freq="D"),
             7,
         ),
         (
             DayOfMonthIndex(),
-            pd.date_range("01-01-2015", periods=365 * 5, freq="D"),
+            pd.period_range("01-01-2015", periods=365 * 5, freq="D"),
             31,
         ),
         (
             DayOfYearIndex(),
-            pd.date_range("01-01-2015", periods=365 * 5, freq="D"),
+            pd.period_range("01-01-2015", periods=365 * 5, freq="D"),
             366,
         ),
         (
             WeekOfYearIndex(),
-            pd.date_range("01-01-2015", periods=53 * 5, freq="W"),
+            pd.period_range("01-01-2015", periods=53 * 5, freq="W"),
             53,
         ),
         (
             MonthOfYearIndex(),
-            pd.date_range("01-01-2015", periods=12 * 5, freq="M"),
+            pd.period_range("01-01-2015", periods=12 * 5, freq="M"),
             12,
         ),
     ],

--- a/test/time_feature/test_lag.py
+++ b/test/time_feature/test_lag.py
@@ -22,6 +22,35 @@ import gluonts.time_feature.lag as date_feature_set
 # Remaining lags correspond to the same `season` (+/- `delta`) in previous `k` cycles.
 expected_lags = {
     # (apart from the default lags) centered around each of the last 3 hours (delta = 2)
+    "4S": [
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        13,
+        14,
+        15,
+        16,
+        17,
+        28,
+        29,
+        30,
+        31,
+        32,
+        43,
+        44,
+        45,
+        46,
+        47,
+        898,
+        899,
+        900,
+        901,
+        902,
+    ],
     "min": [
         1,
         2,
@@ -276,6 +305,7 @@ expected_lags["7D"] = expected_lags["1W"]
 def test_lags():
 
     freq_strs = [
+        "4S",
         "min",
         "1min",
         "15min",

